core-build/gulp-core-build-typescript/src/miniTasks/TslintCmdTask.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as Gulp from 'gulp';
+import * as path from 'path';
+import {
+  JsonFile,
+  IPackageJson
+} from '@microsoft/node-core-library';
+import * as fsx from 'fs-extra';
+import * as TSLint from 'tslint';
+
+import {
+  BaseCmdTask,
+  ICallCmdOptions
+} from './BaseCmdTask';
+
+/**
+ * @public
+ */
+export interface ITslintCmdTaskConfig {
+  /**
+   * The path to the tslint package if the task should override the version of tslint.
+   */
+  tslintPackagePath?: string;
+
+  /**
+   * Optional list of custom args to pass to "tslint"
+   */
+  customArgs?: string[];
+
+  /**
+   * The directory in which tslint should be invoked.
+   */
+  buildDirectory?: string;
+
+  /**
+   * If true, displays warnings as errors. Defaults to false.
+   */
+  displayAsError?: boolean;
+}
+
+/**
+ * @alpha
+ */
+export class TslintCmdTask extends BaseCmdTask<ITslintCmdTaskConfig> {
+  constructor() {
+    super(
+      'tslint',
+      {
+        displayAsError: false
+      }
+    );
+  }
+
+  public loadSchema(): Object {
+    return JsonFile.load(path.resolve(__dirname, '..', 'schemas', 'tslint-cmd.schema.json'));
+  }
+
+  public executeTask(gulp: typeof Gulp, completeCallback: (error?: string) => void): Promise<void> | undefined {
+    // require.resolve('tslint') resolves to <package>/lib/index.ts, and we want the package path
```

Comment:
I wonder if you could move most of the code into the base class (including this function) and just have it call into overriden/abstract functions in the child classes.